doc/api/stream.markdown

```diff
@@ -1315,6 +1315,135 @@ for examples and testing, but there are occasionally use cases where
 it can come in handy as a building block for novel sorts of streams.
 
 
+## Simplified API Via Revealing Constructor Pattern
```

Comment:
Technically I wouldn't call this a revealing constructor pattern, since no functionality is "revealed"; you instead just use `stream.push` and `stream.emit("error", ...)` and similar.